examples/subprojects/project_a/dobi.yaml

```diff
@@ -0,0 +1,22 @@
+#
+# project_a dobi file for subprojects example
+# See README.md for docs
+#
+
+#
+# Images
+#
+#
+
+image=project_a_img:
+    image: example/subprojects/project_a
+    context: project_a
```

Comment:
I think this is one issue. The "build context" is being set to `./project_a`, but `hello_world.sh` is outside of that directory. Instead you can do:

```yaml
context: . 
dockerfile: project_a/Dockerfile
```